src/profile-logic/call-node-info.js

```diff
@@ -201,4 +253,1399 @@ export class CallNodeInfoImpl implements CallNodeInfo {
+    return deepNodes;
+  }
+
+  // This function returns a CallNodePath from a InvertedCallNodeHandle.
+  getCallNodePathFromIndex(
+    callNodeHandle: InvertedCallNodeHandle | null
+  ): CallNodePath {
+    if (callNodeHandle === null || callNodeHandle === -1) {
+      return [];
+    }
+
+    const rootCount = this._rootCount;
+    const callNodePath = [];
+    let currentHandle = callNodeHandle;
+    while (currentHandle >= rootCount) {
+      const nonRootIndex = currentHandle - rootCount;
+      callNodePath.push(this._invertedNonRootCallNodeTable.func[nonRootIndex]);
+      currentHandle = this._invertedNonRootCallNodeTable.prefix[nonRootIndex];
+    }
+    const rootFunc = currentHandle;
+    callNodePath.push(rootFunc);
+    callNodePath.reverse();
+    return callNodePath;
+  }
+
+  // Returns a CallNodeIndex from a CallNodePath.
+  getCallNodeIndexFromPath(
+    callNodePath: CallNodePath
+  ): InvertedCallNodeHandle | null {
+    if (callNodePath.length === 0) {
+      return null;
+    }
+
+    if (callNodePath.length === 1) {
+      return callNodePath[0]; // For roots, IndexIntoFuncTable === InvertedCallNodeHandle
+    }
+
+    const pathDepth = callNodePath.length - 1;
+    let deepestKnownAncestor = this._findDeepestKnownAncestor(callNodePath);
+    let deepestKnownAncestorDepth = this.depthForNode(deepestKnownAncestor);
+
+    while (deepestKnownAncestorDepth < pathDepth) {
+      const currentChildFunc = callNodePath[deepestKnownAncestorDepth + 1];
+      const children = this.getChildren(deepestKnownAncestor);
+      const childMatchingFunc = this._getChildWithFunc(
+        children,
+        currentChildFunc
+      );
+      if (childMatchingFunc === null) {
+        // No child matches the func we were looking for.
+        // This can happen when the provided call path doesn't exist. In that case
+        // we return null.
+        return null;
+      }
+      deepestKnownAncestor = childMatchingFunc;
+      deepestKnownAncestorDepth++;
+    }
+    return deepestKnownAncestor;
+  }
+
+  // Returns the CallNodeIndex that matches the function `func` and whose parent's
+  // CallNodeIndex is `parent`.
+  getCallNodeIndexFromParentAndFunc(
+    parent: InvertedCallNodeHandle | -1,
+    func: IndexIntoFuncTable
+  ): InvertedCallNodeHandle | null {
+    if (parent === -1) {
+      return func; // For roots, IndexIntoFuncTable === InvertedCallNodeHandle
+    }
+    const children = this.getChildren(parent);
+    return this._getChildWithFunc(children, func);
+  }
+
+  _pathHashForNode(callNodeHandle: InvertedCallNodeHandle): string {
+    if (callNodeHandle < this._rootCount) {
+      // callNodeHandle is a root, and for roots, InvertedCallNodeHandle === IndexIntoFuncTable.
+      return hashPathSingleFunc(callNodeHandle);
+    }
+    const nonRootIndex = callNodeHandle - this._rootCount;
+    return this._invertedNonRootCallNodeTable.pathHash[nonRootIndex];
+  }
+
+  prefixForNode(
```

Comment:
prefixForNode doesn't seem covered by tests, but I think this shows a lack of prior tests for inverted trees rather than a problem in this patch. This is called by CallTree.getParent, which is called when the user presses ArrowLeft in the tree.
I think this is out of scope for this patch to add coverage for this.